### PR TITLE
[processor/transform] Added .string accessor for traceId and spanId

### PR DESCRIPTION
**Pull request by davidgargti20**

### CODE REVIEW

**Comment by davidgargti20** [comment]
#11555 
@TylerHelmuth please review this

File: processor/transformprocessor/internal/logs/logs.go
Comment (TylerHelmuth) [inline processor/transformprocessor/internal/logs/logs.go:126]: Please update these cases to use a switch statement like "resource".  Also, please create new accessor functions.  The new accessor functions should return strings in the Getter and accept strings in the Setter.  [This PR contains the old string code for Trace and Span IDs and can be used again.](https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/10903).

**Review by TylerHelmuth — COMMENTED** [review]
The same accessor capabilities should be added to Traces as well.

File: processor/transformprocessor/internal/logs/logs.go
Comment (TylerHelmuth) [reply processor/transformprocessor/internal/logs/logs.go:126]: Correct you'll need to add it back, same with ParseSpanID.  Both functions are in the PR I linked in `parser.go`

File: processor/transformprocessor/internal/common/parser.go
Comment (TylerHelmuth) [inline processor/transformprocessor/internal/common/parser.go:202]:
```suggestion
}

```

**Review by TylerHelmuth — COMMENTED** [review]
@davidgargti20 looks good.  Please also add some unit tests to `logs_test.go` and `traces_test.go` that cover the new accessor scenarios.  Please also add a changelog entry and update the processor's README so it describes how this new `.string` accessor works.

**Comment by davidgargti20** [comment]
> 

@TylerHelmuth what extra test should be included in test as we are already testing that if we can set traceId and spanId or not.

**Comment by davidgargti20** [comment]
@TylerHelmuth added tests

**Review by TylerHelmuth — COMMENTED** [review]
@davidgargti20 Tests and changelog look good.  Last thing to do is update the readme to highlight the new feature.  Please add a new sub-bullet under the "Path expressions" section describing how these new accessors work.  Also please merge in latest from main and handle any conflicts.

**Comment by davidgargti20** [comment]
@TylerHelmuth hopefully all things completed should be good to merge

**Review by TylerHelmuth — APPROVED** [review]
@davidgargti20 thanks!

**Comment by davidgargti20** [comment]
@TylerHelmuth can we have this merged

**Comment by TylerHelmuth** [comment]
@davidgargti20 it needs another person to review it but then it will be merged. Once merged it won't be available in a distribution until 0.55.0.

**Comment by mx-psi** [comment]
@davidgargti20 can you address the linter issues?

```
internal/logs/logs.go:113: File is not `gofmt`-ed with `-s` (gofmt)
		    return accessTraceID(), nil
internal/logs/logs_test.go:18: File is not `gofmt`-ed with `-s` (gofmt)
    	"encoding/hex"
internal/traces/traces.go:94: File is not `gofmt`-ed with `-s` (gofmt)
		    return accessTraceID(), nil
```

**Comment by djaglowski** [comment]
@mx-psi, this is expected because GitHub marks "downstream" jobs as Skipped when any of their upstream jobs fail. For example, `cross-compile` is also marked as Skipped because it needs `unittest`, `integration-tests`, and `lint`, at least one of which did not succeed.

What's really important here is that `lint` is required by branch protection rules, so anything other than success will block the PR.

**Comment by davidgargti20** [comment]
@mx-psi @djaglowski could you approve the workflows so they get started

**Comment by mx-psi** [comment]
@djaglowski Thanks for the explanation!

> What's really important here is that lint is required by branch protection rules, so anything other than success will block the PR.

I believe I was able to click the 'squash and merge' button before, but I may have been confused. If it happens again I will let you know :)

File: processor/transformprocessor/README.md
Comment (davidgargti20) [reply processor/transformprocessor/README.md:21]: @codeboten yes

Comment (davidgargti20) [reply processor/transformprocessor/README.md]: @codeboten corrected typo

Comment (codeboten) [reply processor/transformprocessor/README.md]: Thanks!

**Review by codeboten — APPROVED** [review]

**Comment by davidgargti20** [comment]
@TylerHelmuth could you please re run the tests some tests failed don't know why

**Comment by TylerHelmuth** [comment]
Adding "Skip Changelog" label to avoid having to add a file for the new changelog process for this PR.  @djaglowski for this release cycle (0.55.0) are we going to stick with the existing changelog's Unreleased section and start autogenerating it for 0.56.0?

**Comment by davidgargti20** [comment]
@TylerHelmuth can we merge it now resolved the merge conflicts or are we waiting for some other review.

**Comment by TylerHelmuth** [comment]
@davidgargti20 in the last week there was a change to our changelog process. [Please checkout this section in the CONTRIBUTING guide and follow its steps to add a new changelog entry.](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#adding-a-changelog-entry)

**Comment by TylerHelmuth** [comment]
@codeboten I am on PTO until Thursday, can you keep an eye on this PR and hopefully get it in before Wednesday's release?

**Comment by davidgargti20** [comment]
@codeboten can you review it now added change log ac to new method.

**Comment by TylerHelmuth** [comment]
@davidgargti20 please undo all changes to CHANGELOG.md then we'll be good to go.

**Comment by davidgargti20** [comment]
@TylerHelmuth did that in previous commit.

**Comment by TylerHelmuth** [comment]
@davidgargti20 it might be bc I'm on my phone (I'm on PTO) but it looks to me like the changelog.md file has additions. The PR shouldn't modify changelog.md at all anymore. I'll review everything again on a computer when I'm back. Let me know if I'm wrong about the changelog file.

**Comment by davidgargti20** [comment]
@TylerHelmuth done

**Comment by mx-psi** [comment]
Changelog check is failing... I think you may need e5f988fe240796c2407fa76b1db95a6d17dd1dde? (So just rebase/merge main)

**Comment by davidgargti20** [comment]
@mx-psi merged the main

**Comment by davidgargti20** [comment]
@mx-psi @TylerHelmuth good to merge now?

**Review by mx-psi — APPROVED** [review]

**Comment by davidgargti20** [comment]
Thanks

**Pull request merged**
